file/rotator.go

```diff
@@ -258,9 +258,9 @@ func (r *Rotator) appendToFile() error {
 		return errors.Wrap(err, "failed to append to existing file")
 	}
 	if r.redirectStderr {
-		RedirectStandardError(r.file)
+		err = RedirectStandardError(r.file)
 	}
-	return nil
+	return err
 }
```

Comment:
A couple of things:
1. That changes the semantics because now the error from `RedirectStandardError` can be returned. I'm not sure whether that's desired or not.
2. I suggest an early return or to handle the error inside the `if` statement so the last return is clearly a no-error return.

```go
	if r.redirectStderr {
		return RedirectStandardError(r.file)
	}
	return nil
}
```

Reply:
I really thought it was a bug but you're right, I should not have changed the behaviour. I rolled the change back and use the early return you suggested.